app/controllers/reviews_controller.rb

```diff
@@ -0,0 +1,60 @@
+class ReviewsController < ApplicationController
+  before_action :authenticate_user!, except: [:index, :show]
+
+  def show
+    @review = Review.find(params[:id])
+  end
+
+  def new
+    @pizzeria = Pizzeria.find(params[:pizzeria_id])
+    @review = Review.new
+  end
+
+  def create
+    @pizzeria = Pizzeria.find(params[:pizzeria_id])
+    @review = @pizzeria.reviews.build(review_params)
+    @review.user = current_user
+    if @review.save
+      flash[:notice] = "Review created"
+      redirect_to review_path(@review)
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @review = current_user.reviews.find(params[:id])
+    @pizzeria = @review.pizzeria
+  end
+
+  def update
+    @review = Review.find(params[:id])
```

Comment:
Scope this to `current_user` (similar to what you did in the `edit` action). Then you don't need to use the `correct_user?` method.